# RFM analysis and Pre-clustering process

This notebook contains the analysis of the final RFM table and the data transformation process necessary to achieve a successfull segmentation process in further steps. 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

path = "../data_raw/raw_rfm_metrics.csv"
df_rfm = pd.read_csv(path)

## RFM Table Validation

Before proceeding with transformation and clustering steps, it is essential to validate the structure and quality of the RFM dataset. This includes verifying data types, checking for missing or invalid values, and ensuring that the Recency, Frequency, and Monetary metrics have been correctly computed and are suitable for further analysis.

In [2]:
df_rfm.info()

<class 'pandas.DataFrame'>
RangeIndex: 5878 entries, 0 to 5877
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Recency    5878 non-null   int64  
 1   Frequency  5878 non-null   int64  
 2   Monetary   5878 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 137.9 KB


In [3]:
df_rfm.describe()

,Recency,Frequency,Monetary
count,5878.000000,5878.000000,5878.000000
mean,201.331916,132.600034,2955.810211
std,209.338707,342.190038,14440.288735
min,1.000000,1.000000,2.950000
25%,26.000000,20.000000,342.280000
50%,96.000000,52.000000,867.740000
75%,380.000000,138.000000,2248.305000
max,739.000000,12435.000000,580987.040000


### Insights and conclusion

The RFM dataset contains 5,878 customers with no missing values across all three metrics, confirming data completeness. Additionally, all variables present valid ranges, with no negative values observed, which ensures consistency with the expected definitions of Recency, Frequency, and Monetary.

From a statistical perspective, the summary metrics reveal high variability across customers, particularly in Frequency and Monetary, as indicated by their large standard deviations relative to their means. The wide range between minimum and maximum values further confirms the presence of extreme values and a highly skewed distribution, which is consistent with previous EDA findings.

Overall, the dataset is structurally suitable for the transformation stage. However, the observed variability and skewness reinforce the need for applying distribution transformations and scaling before clustering.